### DISCUSSION TITLE
[Feature] use avif for thumbnails

### DISCUSSION THREAD

**Discussion by chnhi** [immich-app/immich#5117] [Feature Request]
### The feature

avif could avoid jpeg artifacts and better performance with low level quality
high resolution may improve the visibility of screenshot

### Platform

- [X] Server
- [ ] Web
- [x] Mobile

**Comment by benmccann** [comment]
Mobile would need support to decode the avif images. Flutter doesn't support avif natively yet, but there is a plugin available for it: https://github.com/yekeskin/flutter_avif

Supporting avif would also help with ultra HDR: https://github.com/immich-app/immich/discussions/7262#discussioncomment-8551418. And it would be interesting to see if it fixes this issue as well: https://github.com/immich-app/immich/issues/10531

[Only 8% of devices](https://www.scientiamobile.com/av1-codec-hardware-decode-adoption/) currently have avif hardware decoding support. But even without hardware support, using [avif may save battery life](https://news.ycombinator.com/item?id=38654380) by allowing mobile radios to power down sooner even when accounting for increased CPU costs for decoding. However, [jpeg may be rendered faster](https://cloudinary.com/blog/contemplating-codec-comparisons) as it can be decoded in a streaming fashion.

**Reply by OlJohnny** [reply]
+1 for allowing the option to set avif as a preview/thumbnail format

As for the question, why AVIF when we already have WebP? AVIF is supposedly around [20%-30% smaller](https://shortpixel.com/blog/avif-vs-webp/#:~:text=The%20Alliance%20for%20Open%20Media%20created%20AVIF%2C%20the%20latest%20and,%2D30%25%20smaller%20than%20WebP.) for the same perceived quality.
This could result in up to 25% faster load times in immich! (not all traffic is purely raw image data)
Also for at least **two years** now, the [browser support for AVIF](https://caniuse.com/avif) has been excellent.

in regards to the HDR discussion, an option could be added when the source image is in hdr:
- only generate SDR previews & thumbnails (status quo)
- generate SDR & HDR AVIF previews & thumbnails and show the HDR versions on devices explicitly supporting HDR

i think this could help in a lot of cases, and the complicated topic of faithful conversion from various HDR formats to SDR can still be solved at a later point.

**Comment by webysther** [comment]
- AVIF is baseline since a year now: https://caniuse.com/avif
- Some big websites started using avif

Maybe support more grain in the settings, for preview or thumb support for avif in static images solve the problem with lack in support for animated images.

**Comment by craigS89** [comment]
I would Be very appreciative if AVIF could be added to the conversion options in immich. The space saving it offers on large photo collections would be really appreciated

**Comment by floriandotorg** [comment]
Would love to see this!

**Comment by webysther** [comment]
Maybe this can the supersede and even closed by JPEG XL support: https://github.com/immich-app/immich/discussions/12945
